src/modules/deposit/service.ts

```diff
@@ -1,14 +1,17 @@
 import { Injectable } from "@nestjs/common";
 import { InjectRepository } from "@nestjs/typeorm";
-import { Repository } from "typeorm";
+import { Repository, IsNull, Not } from "typeorm";
 import { Deposit } from "../scraper/model/deposit.entity";
 
 @Injectable()
 export class DepositService {
   constructor(@InjectRepository(Deposit) private depositRepository: Repository<Deposit>) {}
 
   public async getDeposits(status = null, limit = 10, offset = 0) {
-    const where = status ? { status } : undefined;
+    const where = {
+      depositDate: Not(IsNull()),
+      status: status || Not(IsNull()),
```

Comment:
This is not a valid TypeORM `or` query. TypeORM allows to add `or` queries by using arrays in the where clause:

```
where: [{ condition1 }, { condition2 }] // where condition1 or condition2
```

Anyway, checking the status column is not needed because it can't be `null`. As you can see [here](https://github.com/across-protocol/scraper-api/blob/stage/src/modules/scraper/model/deposit.entity.ts#L44-L45), it has a default value and it also cannot be `null` because of the NOT NULL restriction coming from [this migration file](https://github.com/across-protocol/scraper-api/blob/stage/migrations/1656106635107-Deposit.ts#L14)

Reply:
So what I try to achieve here is the following
- if `status` param is set, then filter by the given status
- if not, then do not filter by status, i.e. all status that are not null or basically all deposits

I agree that this is not easily understandable. I can rewrite to something like
```js
const baseWhere = { depositDate: Not(IsNull()) };
const where = status ? { ...baseWhere, status } : baseWhere;
```
?

Reply:
it possible that if status is undefined, then the query will contain no references to the status column

Reply:
yes, you were right 👍 much cleaner now and works